src/python/grpcio/grpc/_cython/_cygrpc/metadata.pyx.pxi

```diff
@@ -0,0 +1,64 @@
+# Copyright 2017 gRPC authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import collections
+
+_Metadatum = collections.namedtuple('_Metadatum', ('key', 'value',))
+
+
+cdef void _c_metadata(metadata, size_t *c_count, grpc_metadata **c_metadata):
+ import logging
+ try:
+  if metadata is None:
+    c_count[0] = 0
+    c_metadata[0] = NULL
+  else:
+    metadatum_count = len(metadata)
+    if metadatum_count == 0:
+      c_count[0] = 0
+      c_metadata[0] = NULL
+    else:
+      c_count[0] = metadatum_count
+      c_metadata[0] = <grpc_metadata *>gpr_malloc(
+          metadatum_count * sizeof(grpc_metadata))
+      for index, (key, value) in enumerate(metadata):
+        encoded_key = _encode(key)
+        encoded_value = value if encoded_key[-4:] == b'-bin' else _encode(value)
+        c_metadata[0][index].key = _slice_from_bytes(encoded_key)
+        c_metadata[0][index].value = _slice_from_bytes(encoded_value)
+ except Exception as e:
+  logging.exception('Huh? Metadata was "%s"!', metadata)
+
+
+cdef void _un_c_metadata(int count, grpc_metadata *c_metadata):
```

Comment:
Can we call this delete_c_metadata?  I needed to look at the function to get a sense of what it did.

Reply:
[I thought I had a good joke going](https://www.google.com/search?q=unsee&tbm=isch), but... okay. How's `_release_c_metadata`?